src/proof_of_possession.rs

```diff
@@ -179,6 +179,24 @@ impl ProofOfPossession {
             type_: Some(JWS_TYPE.to_string()),
             ..Default::default()
         };
+        Ok((header, payload))
+    }
+
+    pub fn to_jwt_signing_input(&self) -> Result<Vec<u8>, ConversionError> {
+        use base64::prelude::*;
+
+        let (header, payload) = self.to_unsigned_jwt()?;
+        let json = serde_json::to_string(&header)?;
+        let header = BASE64_URL_SAFE_NO_PAD.encode(json);
+        let signing_input = [header.as_bytes(), b".", payload.as_bytes()]
+            .concat()
+            .to_vec();
+        Ok(signing_input)
+    }
+
+    pub fn to_jwt(&self) -> Result<String, ConversionError> {
```

Comment:
This would benefit from being refactored to use a `Signer` trait, but it can wait!

Reply:
Yeah, Jacob mentioned this about the pop prepare/complete pair of functions in SpruceKit as well.